src/device_utils.cpp

```diff
@@ -36,10 +37,16 @@ str_map get_serial_dev_info()
         "usb",
         "usb_device");
 
-      if (parent_dev)
+      if (parent_dev != nullptr)
       {
-        std::string dev_serial = std::string(udev_device_get_sysattr_value(parent_dev, "serial"));
-        devices[dev_file_name_path] = dev_serial;
+        const auto serial = udev_device_get_sysattr_value(parent_dev, "serial");
+        if (serial) {
+          std::string dev_serial = std::string(serial);
+          devices[dev_file_name_path] = dev_serial;
+        }
+        else {
+          ROS_WARN_STREAM("Found device with null serial: '" << dev_file_name_path << "'");
```

Comment:
Is this always going to print null or an empty string?

Reply:
no, its the `serial` variable that will be null, the `dev_file_name_path` var will have a good value like `/dev/video8`